addons/project/models/project_task.py

```diff
@@ -1590,3 +1595,16 @@ def action_redirect_to_project_task_form(self):
             'url': '/web#model=project.task&id=%s&action=%s&view_type=form' % (self.id, self.env.ref('project.action_view_my_task').id),
             'target': 'new',
         }
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True): #TODO: multiple groupby ??
+        # A read_group can not be performed if records are grouped by personal_stage_type_id as it is a computed field.
+        # personal_stage_type_ids behaves like a M2O from the point of view of the user, we therefore use this field instead.
+        if 'personal_stage_type_id' in groupby and (not lazy or groupby[0] == 'personal_stage_type_id'):
+            groupby = list(map(lambda field: "personal_stage_type_ids" if field == "personal_stage_type_id" else field, groupby)) # limitation, problem when grouped on personal_stage_type_id and personal_stage_type_ids but who does that?
```

Comment:
```suggestion
            groupby = ["personal_stage_type_ids" if field == "personal_stage_type_id" else field for field in groupby] # limitation, problem when grouped on personal_stage_type_id and personal_stage_type_ids but who does that?
```
I think it is more pythonian